scripts/bump_version.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Get the path of the script
+SCRIPT_PATH=$(dirname $(realpath -s $0))
+
+####################
+# Check preconditions
+####################
+
+# Check for a version number passed by argument
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 version-number"
+    exit 1
+fi
+
+# Validate version - source this so we get the version information
+source ${SCRIPT_PATH}/validate_version.sh $1
+if [ $? -ne 0 ]; then
+    exit 1
+fi
+
+####################
+# Bump version
+####################
+
+# Update version in the CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_MAJOR \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_MAJOR \"$MAJOR_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_MINOR \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_MINOR \"$MINOR_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (CPACK_PACKAGE_VERSION_PATCH \"[0-9]\+\")/set (CPACK_PACKAGE_VERSION_PATCH \"$PATCH_VERSION\")/g" $SCRIPT_PATH/../CMakeLists.txt
+sed -i "s/set (VERSION_SUFFIX \"[^\"]*\")/set (VERSION_SUFFIX \"$VERSION_SUFFIX\")/g" ${SCRIPT_PATH}/../CMakeLists.txt
```

Comment:
We set `PROJECT_VERSION` using `CPACK_PACKAGE_VERSION_*`. https://github.com/xournalpp/xournalpp/blob/fffaa72e432a529cee7d343f0e337f491d98b431/CMakeLists.txt#L12

This seems appropriate to me since we actually care about the major/minor/patch versions.

Reply:
Since the behaviour of CPack changed to:
[```
CPACK_PACKAGE_VERSION_MAJOR

    Package major version. This variable will always be set, but its default value depends on whether or not version details were given to the project() command in the top level CMakeLists.txt file. If version details were given, the default value will be CMAKE_PROJECT_VERSION_MAJOR. If no version details were given, a default version of 0.1.1 will be assumed, leading to CPACK_PACKAGE_VERSION_MAJOR having a default value of 0.

```](https://cmake.org/cmake/help/v3.21/module/CPack.html#variable:CPACK_PACKAGE_VERSION_MAJOR)

Reply:
As long as the version is written somewhere I can parse it from...

I will update this logic once #2570 is merged.